src/Microsoft.AspNet.Razor/Parser/TokenizerBackedParser.cs

```diff
@@ -75,6 +76,39 @@ protected void Initialize(SpanBuilder span)
             }
         }
 
+        protected TSymbol Lookahead(int count = 1)
```

Comment:
It does :smile:. When EOF is hit it starts pushing `null`s into the `symbols` array, essentially it returns null when it hits EOF which is what every other method does as well.